[BUG] openapi: SIGSEGV on 'application/json: {}'

### Is there an existing issue for this?

- [x] I have searched the existing issues.

### Current Behavior

Nuclei crashes with SIGSEGV when openapi spec contains
```
content: 
  application/json: {}
```
Verified that switching b/w yaml & xml makes no difference.

### Expected Behavior

Nuclei should treat this as empty content.

### Steps To Reproduce

Minimal openapi.yml:
```
openapi: 3.1.0
servers:
- url: http://localhost:8080
paths:
  /sigsegv:
    get:
      requestBody:
        content:
          application/json: {}
```

Running `nuclei -im openapi -l openapi.yml` produces:

```

                     __     _
   ____  __  _______/ /__  (_)
  / __ \/ / / / ___/ / _ \/ /
 / / / / /_/ / /__/ /  __/ /
/_/ /_/\__,_/\___/_/\___/_/   v3.3.9

		projectdiscovery.io

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x18 pc=0x1012540ac]

goroutine 1 [running]:
github.com/projectdiscovery/nuclei/v3/pkg/input/formats/openapi.generateRequestsFromOp(0x140006d3ea0)
	github.com/projectdiscovery/nuclei/v3/pkg/input/formats/openapi/generator.go:271 +0xf8c
github.com/projectdiscovery/nuclei/v3/pkg/input/formats/openapi.GenerateRequestsFromSchema(0x14000809180, {0x1400052eb10?, 0x40?, 0xeb?}, 0x14000f55c30)
	github.com/projectdiscovery/nuclei/v3/pkg/input/formats/openapi/generator.go:95 +0xac4
github.com/projectdiscovery/nuclei/v3/pkg/input/formats/openapi.(*OpenAPIFormat).Parse(0x105765e60, {0x103d116c0, 0x1400052eb40}, 0x14000f55c30, {0x103d111c0?, 0x1053220a0?})
	github.com/projectdiscovery/nuclei/v3/pkg/input/formats/openapi/openapi.go:40 +0x9c
github.com/projectdiscovery/nuclei/v3/pkg/input/provider/http.NewHttpInputProvider(0x140012078b0)
	github.com/projectdiscovery/nuclei/v3/pkg/input/provider/http/multiformat.go:88 +0x374
github.com/projectdiscovery/nuclei/v3/pkg/input/provider.NewInputProvider({0x105775680?, 0x0?})
	github.com/projectdiscovery/nuclei/v3/pkg/input/provider/interface.go:112 +0x23c
github.com/projectdiscovery/nuclei/v3/internal/runner.New(0x105775680)
	github.com/projectdiscovery/nuclei/v3/internal/runner/runner.go:243 +0x840
main.main()
	github.com/projectdiscovery/nuclei/v3/cmd/nuclei/main.go:153 +0x3e0
```

### Relevant log output

```shell

```

### Environment

```markdown
- OS: OSX 15.0
- Nuclei: 3.3.9
- Go: N/A
```

### Anything else?

_No response_

--- a/pkg/input/formats/openapi/examples.go
+++ b/pkg/input/formats/openapi/examples.go
@@ -288,3 +288,33 @@ func openAPIExample(schema *openapi3.Schema, cache map[*openapi3.Schema]*cachedS
 func generateExampleFromSchema(schema *openapi3.Schema) (interface{}, error) {
 	return openAPIExample(schema, make(map[*openapi3.Schema]*cachedSchema)) // TODO: Use caching
 }
+
+func generateEmptySchemaValue(contentType string) *openapi3.Schema {
+	schema := &openapi3.Schema{}
+	objectType := &openapi3.Types{"object"}
+	stringType := &openapi3.Types{"string"}
+
+	switch contentType {
+	case "application/json":
+		schema.Type = objectType
+		schema.Properties = make(map[string]*openapi3.SchemaRef)
+	case "application/xml":
+		schema.Type = stringType
+		schema.Format = "xml"
+		schema.Example = "<?xml version=\"1.0\"?><root/>"
+	case "text/plain":
+		schema.Type = stringType
+	case "application/x-www-form-urlencoded":
+		schema.Type = objectType
+		schema.Properties = make(map[string]*openapi3.SchemaRef)
+	case "multipart/form-data":
+		schema.Type = objectType
+		schema.Properties = make(map[string]*openapi3.SchemaRef)
+	case "application/octet-stream":
+	default:
+		schema.Type = stringType
+		schema.Format = "binary"
+	}
+
+	return schema
+}

--- a/pkg/input/formats/openapi/generator.go
+++ b/pkg/input/formats/openapi/generator.go
@@ -268,24 +268,32 @@ func generateRequestsFromOp(opts *generateReqOptions) error {
 		for content, value := range opts.op.RequestBody.Value.Content {
 			cloned := req.Clone(req.Context())
 
-			example, err := generateExampleFromSchema(value.Schema.Value)
-			if err != nil {
-				continue
+			var val interface{}
+
+			if value.Schema == nil || value.Schema.Value == nil {
+				val = generateEmptySchemaValue(content)
+			} else {
+				var err error
+
+				val, err = generateExampleFromSchema(value.Schema.Value)
+				if err != nil {
+					continue
+				}
 			}
 
 			// var body string
 			switch content {
 			case "application/json":
-				if marshalled, err := json.Marshal(example); err == nil {
+				if marshalled, err := json.Marshal(val); err == nil {
 					// body = string(marshalled)
 					cloned.Body = io.NopCloser(bytes.NewReader(marshalled))
 					cloned.ContentLength = int64(len(marshalled))
 					cloned.Header.Set("Content-Type", "application/json")
 				}
 			case "application/xml":
-				exampleVal := mxj.Map(example.(map[string]interface{}))
+				values := mxj.Map(val.(map[string]interface{}))
 
-				if marshalled, err := exampleVal.Xml(); err == nil {
+				if marshalled, err := values.Xml(); err == nil {
 					// body = string(marshalled)
 					cloned.Body = io.NopCloser(bytes.NewReader(marshalled))
 					cloned.ContentLength = int64(len(marshalled))
@@ -294,7 +302,7 @@ func generateRequestsFromOp(opts *generateReqOptions) error {
 					gologger.Warning().Msgf("openapi: could not encode xml")
 				}
 			case "application/x-www-form-urlencoded":
-				if values, ok := example.(map[string]interface{}); ok {
+				if values, ok := val.(map[string]interface{}); ok {
 					cloned.Form = url.Values{}
 					for k, v := range values {
 						cloned.Form.Set(k, types.ToString(v))
@@ -306,7 +314,7 @@ func generateRequestsFromOp(opts *generateReqOptions) error {
 					cloned.Header.Set("Content-Type", "application/x-www-form-urlencoded")
 				}
 			case "multipart/form-data":
-				if values, ok := example.(map[string]interface{}); ok {
+				if values, ok := val.(map[string]interface{}); ok {
 					buffer := &bytes.Buffer{}
 					multipartWriter := multipart.NewWriter(buffer)
 					for k, v := range values {
@@ -326,13 +334,13 @@ func generateRequestsFromOp(opts *generateReqOptions) error {
 					cloned.Header.Set("Content-Type", multipartWriter.FormDataContentType())
 				}
 			case "text/plain":
-				str := types.ToString(example)
+				str := types.ToString(val)
 				// body = str
 				cloned.Body = io.NopCloser(strings.NewReader(str))
 				cloned.ContentLength = int64(len(str))
 				cloned.Header.Set("Content-Type", "text/plain")
 			case "application/octet-stream":
-				str := types.ToString(example)
+				str := types.ToString(val)
 				if str == "" {
 					// use two strings
 					str = "string1\nstring2"

